core/src/test/java/io/undertow/server/security/ClientCertRenegotiationTestCase.java

```diff
@@ -67,6 +68,8 @@ protected List<AuthenticationMechanism> getTestMechanisms() {
 
     @BeforeClass
     public static void startSSL() throws Exception {
+        Assume.assumeTrue("UNDERTOW-2112 New version TLSv1.3 and JDK14 and newer versions are breaking this feature",
+                getJavaSpecificationVersion() < 14);
```

Comment:
I believe TLSv1.3 was backported to jdk8 a, it may instead be preferable to explicitly specify TLSv1.2 for the server in this test, which should work across java versions

Reply:
@carterkozak TLSV1.2 is already used in the test but it fails with JDK-17. The problem is that JDK-14+ added a new ticket extension that makes the `force_renegotiation` fail in TLSv1.2. Mainly all new versions (TLS spec of JDK impl) are making `force_renegotiation` fail. See [UNDERTOW-2112](https://issues.redhat.com/browse/UNDERTOW-2112) for more information. The test should be assumed to avoid it in jdk-14+.